client-course-schedulizer/src/readCSV.ts

```diff
@@ -0,0 +1,428 @@
+import papa from "papaparse";
+import * as di from "./dataInterfaces";
+
+// For now the plan is to assume the CSV has these proper headers
+const validFields = [
+  "name",
+  // If both are found, use "prefixes"
+  "prefix",
+  "prefixes",
+  "number",
+  // Corresponds to letter
+  "section",
+  "studentHours",
+  "facultyHours",
+  // If both are found, use "startTimeStr"
+  "startTime",
+  "startTimeStr",
+  "duration",
+  // Covers both building and room number
+  "location",
+  "roomCapacity",
+  "year",
+  "term",
+  "half",
+  // Needs to be parsed
+  "days",
+  "globalMax",
+  "localMax",
+  "anticipatedSize",
+  // If both are found, use "instructors"
+  "instructor",
+  "instructors",
+  "comments",
+];
+
+// Define regexes for parsing
+const timeReg = RegExp("(?<![1-9])(1[0-9]|2[0-3]|[0-9]):([0-5][0-9])");
+const amReg = RegExp("[Aa][Mm]");
+const pmReg = RegExp("[Pp][Mm]");
+const fallReg = RegExp("[Ff]");
+const summerReg = RegExp("[Ss][Uu]");
+const springReg = RegExp("[Ss]");
+// "W" represents interim in Pruim's system it seems
+const interimReg = RegExp("[Ii]|[Ww]");
+const firstReg = RegExp("[Ff]irst");
+const secondReg = RegExp("[Ss]econd");
+const fullReg = RegExp("[Ff]ull");
+const sunReg = RegExp("[Ss][Uu]|[Nn]");
+const monReg = RegExp("[Mm]");
+const tuesReg = RegExp("[Tt](?![Hh])");
+const wedReg = RegExp("[Ww]");
+const thursReg = RegExp("[Tt][Hh]|[Rr]");
+const friReg = RegExp("[Ff]");
+const satReg = RegExp("[Ss](?![Uu])");
+
+const convertToInterface = function convertToInterface(objects: papa.ParseResult<never>) {
+  // Define variables for Schedule creation
+  let name: string;
+  let prefixes: string[];
+  let number: string;
+  let letter: string;
+  let studentHours: number;
+  let facultyHours: number;
+  let hourPart: string;
+  let numHourPart: number;
+  let minPart: string;
+  let ampm: string;
+  let startTime: string;
+  let duration: number;
+  let building: string;
+  let roomNumber: string;
+  let roomCapacity: number;
+  let year: number;
+  let term: di.Term;
+  let half: di.Half;
+  let days: di.Day[];
+  let globalMax: number;
+  let localMax: number;
+  let anticipatedSize: number;
+  let instructors: di.Instructor[];
+  let comments: string;
+  // eslint-disable-next-line prefer-const
+  const schedule: di.Schedule = {
+    sections: [],
+  };
+
+  // Get data and fields from the CSV
+  const { data, meta } = objects;
+  const { fields } = meta;
+
+  // From the CSV fields, take the ones which we recognize
+  // eslint-disable-next-line prefer-const
+  let usableFields: string[] = [];
+  let field;
+  let startTimeIdx = -1;
+  let startTimeStrIdx = -1;
+  let instructorIdx = -1;
+  let instructorsIdx = -1;
+  let prefixIdx = -1;
+  let prefixesIdx = -1;
```

Comment:
I believe you are able to do chained assignment, like `let startTimeIdx = startTimeStrIdx = -1;` but will all of the values. Could save some lines (but not a major issue; more of a note).

Reply:
`eslint` is setup to get mad at multi-assignment, so I think I'll leave it as is for now. (For reference the rule is `no-multi-assign`)